packages/react-components/react-combobox/src/components/Option/useOption.tsx

```diff
@@ -34,21 +35,41 @@ export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElemen
   const optionRef = React.useRef<HTMLElement>(null);
   const optionValue = getValueString(value, props.children);
 
+  // use the id if provided, otherwise use a generated id
+  const defaultId = useId('fluent-option');
+  const id = React.useMemo(() => {
+    return props.id || defaultId;
+  }, [props.id, defaultId]);
+
+  // data used for context registration & events
+  const optionData = React.useMemo<OptionValue>(() => ({ id, disabled, value: optionValue }), [
+    id,
+    disabled,
+    optionValue,
+  ]);
+
   // context values
+  const setOpen = useContextSelector(ComboboxContext, ctx => ctx.setOpen);
   const multiselect = useContextSelector(ListboxContext, ctx => ctx.multiselect);
-  const onOptionClick = useContextSelector(ListboxContext, ctx => ctx.onOptionClick);
   const registerOption = useContextSelector(ListboxContext, ctx => ctx.registerOption);
   const selected = useContextSelector(ListboxContext, ctx => {
     const selectedOptions = ctx.selectedOptions;
 
     return !!optionValue && !!selectedOptions.find(o => o === optionValue);
   });
 
-  // use the id if provided, otherwise use a generated id
-  const defaultId = useId('fluent-option');
-  const id = React.useMemo(() => {
-    return props.id || defaultId;
-  }, [props.id, defaultId]);
+  const onOptionClick = useContextSelector(ListboxContext, ctx => {
```

Comment:
I think `useContextSelector` is meant to simply select a value out of the context. More complex things should be done with the resulting values instead:
```
const setActiveOption = useContextSelector(ListboxContext, ctx => ctx.setActiveOption);
const selectOption = useContextSelector(ListboxContext, ctx => ctx.selectOption);
const onOptionClick = (ev) => { ... };